internal/pkg/controllers_metrics/thanosquery_metrics.go

```diff
@@ -0,0 +1,26 @@
+package controllers_metrics
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type ThanosQueryMetrics struct {
+	*BaseMetrics
+	EndpointsConfigured                *prometheus.GaugeVec
+	ServiceWatchesReconciliationsTotal prometheus.Counter
+}
+
+func NewThanosQueryMetrics(reg prometheus.Registerer) ThanosQueryMetrics {
```

Comment:
Let's just pass it in directly after initializing once in main.go?
```suggestion
func NewThanosQueryMetrics(baseMetrics *BaseMetrics, reg prometheus.Registerer) ThanosQueryMetrics {
```